apps/setuichange/boot.js

```diff
@@ -0,0 +1,156 @@
+  }
+  if (Bangle.btnWatches) {
+    Bangle.btnWatches.forEach(clearWatch);
+    delete Bangle.btnWatches;
+  }
+  if (Bangle.swipeHandler) {
+    Bangle.removeListener("swipe", Bangle.swipeHandler);
+    delete Bangle.swipeHandler;
+  }
+  if (Bangle.dragHandler) {
+    Bangle.removeListener("drag", Bangle.dragHandler);
+    delete Bangle.dragHandler;
+  }
+  if (Bangle.touchHandler) {
+    Bangle.removeListener("touch", Bangle.touchHandler);
+    delete Bangle.touchHandler;
+  }
+  delete Bangle.uiRedraw;
+  delete Bangle.CLOCK;
+  if (Bangle.uiRemove) {
+    let r = Bangle.uiRemove;
+    delete Bangle.uiRemove; // stop recursion if setUI is called inside uiRemove
+    r();
+  }
+  g.reset();// reset graphics state, just in case
+  if (!mode) return;
+  function b() {
+    try{Bangle.buzz(30);}catch(e){}
+  }
+  if (mode=="updown") {
+    var dy = 0;
+    Bangle.dragHandler = e=>{
+      dy += e.dy;
+      if (!e.b) dy=0;
+      while (Math.abs(dy)>32) {
+        if (dy>0) { dy-=32; cb(1) }
+        else { dy+=32; cb(-1) }
+        Bangle.buzz(20);
+      }
+    };
+    Bangle.on('drag',Bangle.dragHandler);
+    Bangle.touchHandler = d => {b();cb();};
+    Bangle.btnWatches = [
+      setWatch(function() { b();cb(); }, BTN1, {repeat:1, edge:"rising"}),
+    ];
+  } else if (mode=="leftright") {
+    var dx = 0;
+    Bangle.dragHandler = e=>{
+      dx += e.dx;
+      if (!e.b) dx=0;
+      while (Math.abs(dx)>32) {
+        if (dx>0) { dx-=32; cb(1) }
+        else { dx+=32; cb(-1) }
+        Bangle.buzz(20);
+      }
+    };
+    Bangle.on('drag',Bangle.dragHandler);
+    Bangle.touchHandler = d => {b();cb();};
+    Bangle.btnWatches = [
+      setWatch(function() { b();cb(); }, BTN1, {repeat:1, edge:"rising"}),
+    ];
+  } else if (mode=="clock") {
+    Bangle.CLOCK=1;
+    Bangle.btnWatches = [
+      setWatch(Bangle.showLauncher, BTN1, {repeat:1,edge:"rising"})
+    ];
+  } else if (mode=="clockupdown") {
+    Bangle.CLOCK=1;
+    Bangle.touchHandler = (d,e) => {
+      if (e.x < 120) return;
+      b();cb((e.y > 88) ? 1 : -1);
+    };
+    Bangle.btnWatches = [
+      setWatch(Bangle.showLauncher, BTN1, {repeat:1,edge:"rising"})
+    ];
+  } else if (mode=="custom") {
+    if (options.clock) {
+      Bangle.btnWatches = [
+        setWatch(Bangle.showLauncher, BTN1, {repeat:1,edge:"rising"})
+      ];
+    }
+  } else
+    throw new Error("Unknown UI mode "+E.toJS(mode));
+  if (options.clock) Bangle.CLOCK=1;
+  if (options.touch)
+  Bangle.touchHandler = options.touch;
+  if (options.drag) {
+    Bangle.dragHandler = options.drag;
+    Bangle.on("drag", Bangle.dragHandler);
+  }
+  if (options.swipe) {
+    Bangle.swipeHandler = options.swipe;
+    Bangle.on("swipe", Bangle.swipeHandler);
+  }
+  if (options.btn || options.btnRelease) {
+    // Make sure we only add the custom btn handlers. TODO: Deliberate on whether this is the right approach or not.
+    Bangle.btnWatches.forEach(clearWatch);
```

Comment:
Do we need this? Or if we do, we need to check Bangle.btnWatches exists as it may not

Reply:
I added it at one point because it seemed like multiple handlers were added without it. But trying without it now it seems to make no difference. Lets try without it - I'll take it out for now.